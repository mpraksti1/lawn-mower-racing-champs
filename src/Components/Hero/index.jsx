import React from 'react';
import PropTypes from 'prop-types';
import Centered from '../../Elements/Centered';
import StyledDiv from './styles';

const Hero = (props) => {
  const { imgUrl, children } = props;

  return (
    <StyledDiv imgUrl={imgUrl}>
      <Centered>
        { children }
      </Centered>
    </StyledDiv>
  );
};

Hero.defaultProps = {
  imgUrl: '',
};

Hero.propTypes = {
  imgUrl: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Hero;
