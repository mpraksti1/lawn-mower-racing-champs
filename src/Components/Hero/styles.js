import styled from 'styled-components';

const StyledDiv = styled.div`
  height: 500px;
  position: relative;
  margin: 0 -${({ theme }) => theme.baseSizeUnit * 2}px;
  background-size: cover;
  background-image: url('/assets/images/homebg.jpg');
`;

export default StyledDiv;
