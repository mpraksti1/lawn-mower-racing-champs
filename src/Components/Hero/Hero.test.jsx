import Hero from '.';

describe('Hero component', () => {
  it('renders without crashing', () => {
    shallow(
      <Hero>
        <p>Hero!</p>
      </Hero>,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <Hero imgUrl="http://via.placeholder.com/350x150">
        <p>Hero!</p>
      </Hero>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
