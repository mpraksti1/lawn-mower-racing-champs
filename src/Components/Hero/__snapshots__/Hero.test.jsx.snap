// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component matches snapshot 1`] = `
<styled.div
  imgUrl="http://via.placeholder.com/350x150"
>
  <Centered>
    <p>
      Hero!
    </p>
  </Centered>
</styled.div>
`;
